--- a/qutebrowser/browser/webengine/webenginetab.py
+++ b/qutebrowser/browser/webengine/webenginetab.py
@@ -1072,11 +1072,11 @@
         script.setWorldId(world)
         script.setRunsOnSubFrames(subframes)
         script.setName(f'_qute_{name}')
-        self._widget.page().scripts().insert(script)
+        self._widget.page().profile().scripts().insert(script)
 
     def _remove_js(self, name):
         """Remove an early QWebEngineScript."""
-        scripts = self._widget.page().scripts()
+        scripts = self._widget.page().profile().scripts()
         if machinery.IS_QT6:
             for script in scripts.find(f'_qute_{name}'):
                 scripts.remove(script)
@@ -1123,7 +1123,7 @@
         self._inject_greasemonkey_scripts(scripts)
 
     def _remove_all_greasemonkey_scripts(self):
-        page_scripts = self._widget.page().scripts()
+        page_scripts = self._widget.page().profile().scripts()
         for script in page_scripts.toList():
             if script.name().startswith("GM-"):
                 log.greasemonkey.debug('Removing script: {}'
@@ -1145,7 +1145,7 @@
         # make sure we replace existing scripts, not just add new ones.
         # While, taking care not to remove any other scripts that might
         # have been added elsewhere, like the one for stylesheets.
-        page_scripts = self._widget.page().scripts()
+        page_scripts = self._widget.page().profile().scripts()
         self._remove_all_greasemonkey_scripts()
 
         seen_names = set()
